src/pages/Browse/index.tsx

```diff
@@ -103,6 +106,11 @@ export class BrowseBase extends React.Component<Props> {
       return;
     }
 
+    if (version && !currentVersionId) {
+      dispatch(versionsActions.setCurrentVersionId({ versionId: version.id }));
+      return;
```

Comment:
This shouldn't return because the same `loadData()` execution may need to also [fetch a file](https://github.com/mozilla/addons-code-manager/blob/5b757e1f6084fbab36891c8edb9b7ec2c88149ac/src/pages/Browse/index.tsx#L116).

I'm assuming you added the return to make testing easier? If that's the case, maybe looking for a specific action would help? Let me know if you get stuck. Example:

https://github.com/mozilla/addons-code-manager/blob/c2b1d9469a96278c29310ad8389c526275bf8304/src/reducers/versions.spec.tsx#L948-L950

Reply:
I thought `_setVersionFile` will be dispatched twice without 'return' here. Now I know I was wrong about it. I updated this patch.

Reply:
Generally, the only reasons to return within a `loadData` function would be when some data you're depending on hasn't loaded yet or if there was an error that would make it impossible to continue dispatching ([example from addons-frontend](https://github.com/mozilla/addons-frontend/blob/3e32871297d43702daa03f0bee03841b175ef789/src/amo/pages/AddonReviewList/index.js#L118-L121)).

> I thought `_setVersionFile` will be dispatched twice without 'return' here. 

Yeah, the code there would probably get executed twice but it should only trigger one dispatch because of how the code is checking the state. In other words, it should be [idempotent](https://en.wikipedia.org/wiki/Idempotence).